core/src/main/java/com/linecorp/armeria/common/util/Exceptions.java

```diff
@@ -222,6 +222,38 @@ public static boolean isStreamCancelling(Throwable cause) {
         return null; // Never reaches here.
     }
 
+    // This is copied from
+    // https://github.com/ReactiveX/RxJava/blob/v3.0.0/src/main/java/io/reactivex/rxjava3/exceptions/Exceptions.java
+
+    /**
+     * Throws a particular {@code Throwable} only if it belongs to a set of "fatal" error varieties. These
+     * varieties are as follows:
+     * <ul>
+     * <li>{@code VirtualMachineError}</li>
+     * <li>{@code ThreadDeath}</li>
+     * <li>{@code LinkageError}</li>
+     * </ul>
+     * This can be useful if you are writing an operator that calls user-supplied code, and you want to
+     * notify subscribers of errors encountered in that code by calling their {@code onError} methods, but only
+     * if the errors are not so catastrophic that such a call would be futile, in which case you simply want to
+     * rethrow the error.
+     *
+     * @param t
+     *         the {@code Throwable} to test and perhaps throw
+     * @see <a href="https://github.com/ReactiveX/RxJava/issues/748#issuecomment-32471495">
+     *     RxJava: StackOverflowError is swallowed (Issue #748)</a>
+     */
+    public static void throwIfFatal(Throwable t) {
+        requireNonNull(t, "t");
```

Comment:
I'm not strong. I think we don't need to check null here. Because a user might not expect NPE here.

Reply:
When we provide a public API, we always don't want a user to call the API with `null`. But we always have a null check. 😄

Reply:
OK, that is a good point, let keep it.